src/esp/assets/ResourceManager.h

```diff
@@ -337,13 +436,63 @@ class ResourceManager {
   int getNumLibraryObjects() { return physicsObjTemplateLibrary_.size(); };
 
   /**
-   * @brief Retrieve the composition of all transforms applied to a mesh since
-   * it was loaded.
+   * @brief Gets the number of loaded object templates stored in the @ref
+   * physicsObjTemplateLibrary_.
+   *
+   * @return The number of entries in @ref physicsObjTemplateLibrary_ that are
+   * loaded from files.
+   */
+  int getNumFileTemplateObjects() { return physicsObjTmpltLibByID_.size(); };
+  /**
+   * @brief Get a random loaded attribute handle for the loaded file-based
+   * object templates
+   *
+   * @return a randomly selected handle corresponding to the a file-based object
+   * attributes template, or empty string if none loaded
+   */
+  std::string getRandomFileTemplateHandle() {
+    return getRandomTemplateHandle(physicsObjTmpltLibByID_, "file-based");
+  }
+
+  /**
+   * @brief Gets the number of primitive template objects stored in the @ref
+   * physicsObjTemplateLibrary_.
+   *
+   * @return The number of entries in @ref physicsObjTemplateLibrary_ that
+   * describe primitives.
+   */
+  int getNumPrimTemplateObjects() { return physicsPrimTmpltLibByID_.size(); };
+  /**
+   * @brief Get a random loaded attribute handle for the loaded primitive object
+   * templates
+   *
+   * @return a randomly selected handle corresponding to the a primitive
+   * attributes template, or empty string if none loaded
+   */
+  std::string getRandomPrimTemplateHandle() {
+    return getRandomTemplateHandle(physicsPrimTmpltLibByID_, "primitive");
+  }
+
+ private:
+  /**
+   * @brief return a random handle selected from the passed map - is passed
+   * either map of ids->prim handles or ids->file obj template handles
+   *
+   * @return a random template handle of the chosen type, or the empty string
+   * if none loaded
+   */
+  std::string getRandomTemplateHandle(std::map<int, std::string>& mapOfHandles,
```

Comment:
does type refer to prim vs. obj here? What is the use case of this?

Reply:
This is the underlying functionality to find the random entry within the partitioned map.  The map that is passed is the specified type of object we wish to have a random sample of.

Reply:
the "type" being passed is for error display only.